File: 3396-valid-word/valid-word.cpp
class Solution {
public:
    bool isVowel(char ch)
    {
        ch = tolower(ch);
        return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
    }

    bool isDigit(char ch)
    {
        return ch >= '0' && ch <= '9';
    }

    bool isValid(string word) 
    {
        if(word.length() < 3)
        return false;

        bool hasConsonant = false, hasVowel = false;
        for(char ch : word)
        {
            if(!isalnum(ch))
            return false;

            else if(isVowel(ch))
            hasVowel = true;

            else if(!isDigit(ch))
            hasConsonant = true;
        }
        return hasConsonant && hasVowel;
    }
};

File: 3396-valid-word/valid-word.java
class Solution {
    public boolean isVowel(char ch) {
        ch = Character.toLowerCase(ch);
        return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
    }

    public boolean isValid(String word) {
        if (word.length() < 3)
            return false;

        boolean hasConsonant = false, hasVowel = false;

        for (char ch : word.toCharArray()) {
            if (!Character.isLetterOrDigit(ch))
                return false;

            else if (isVowel(ch))
                hasVowel = true;

            else if (!Character.isDigit(ch))
                hasConsonant = true;
        }

        return hasConsonant && hasVowel;
    }
}
